examples/material_gallery/lib/demo/weathers_demo.dart

```diff
@@ -0,0 +1,495 @@
+    _addParticles(2.0);
+  }
+
+  List<ParticleSystem> _particles = [];
+
+  void _addParticles(double distance) {
+    ParticleSystem particles = new ParticleSystem(
+      _sprites['raindrop.png'],
+      transferMode: TransferMode.srcATop,
+      posVar: new Point(1300.0, 0.0),
+      direction: 90.0,
+      directionVar: 0.0,
+      speed: 1000.0 / distance,
+      speedVar: 100.0 / distance,
+      startSize: 1.2 / distance,
+      startSizeVar: 0.2 / distance,
+      endSize: 1.2 / distance,
+      endSizeVar: 0.2 / distance,
+      life: 1.5 * distance,
+      lifeVar: 1.0 * distance
+    );
+    particles.position = new Point(1024.0, -200.0);
+    particles.rotation = 10.0;
+    particles.opacity = 0.0;
+
+    _particles.add(particles);
+    addChild(particles);
+  }
+
+  set active(bool active) {
+    actions.stopAll();
+    for (ParticleSystem system in _particles) {
+      if (active) {
+        actions.run(
+          new ActionTween((a) => system.opacity = a, system.opacity, 1.0, 2.0
+        ));
+      } else {
+        actions.run(
+          new ActionTween((a) => system.opacity = a, system.opacity, 0.0, 0.5
+        ));
+      }
+    }
+  }
+}
+
+class Snow extends Node {
+  Snow() {
+    _addParticles(_sprites['flake-0.png'], 1.0);
+    _addParticles(_sprites['flake-1.png'], 1.0);
+    _addParticles(_sprites['flake-2.png'], 1.0);
+
+    _addParticles(_sprites['flake-3.png'], 1.5);
+    _addParticles(_sprites['flake-4.png'], 1.5);
+    _addParticles(_sprites['flake-5.png'], 1.5);
+
+    _addParticles(_sprites['flake-6.png'], 2.0);
+    _addParticles(_sprites['flake-7.png'], 2.0);
+    _addParticles(_sprites['flake-8.png'], 2.0);
+  }
+
+  List<ParticleSystem> _particles = [];
+
+  void _addParticles(Texture texture, double distance) {
+    ParticleSystem particles = new ParticleSystem(
+      texture,
+      transferMode: TransferMode.srcATop,
+      posVar: new Point(1300.0, 0.0),
+      direction: 90.0,
+      directionVar: 0.0,
+      speed: 150.0 / distance,
+      speedVar: 50.0 / distance,
+      startSize: 1.0 / distance,
+      startSizeVar: 0.3 / distance,
+      endSize: 1.2 / distance,
+      endSizeVar: 0.2 / distance,
+      life: 20.0 * distance,
+      lifeVar: 10.0 * distance,
+      emissionRate: 2.0,
+      startRotationVar: 360.0,
+      endRotationVar: 360.0,
+      radialAccelerationVar: 10.0 / distance,
+      tangentialAccelerationVar: 10.0 / distance
+    );
+    particles.position = new Point(1024.0, -50.0);
+    particles.opacity = 0.0;
+
+    _particles.add(particles);
+    addChild(particles);
+  }
+
+  set active(bool active) {
+    actions.stopAll();
+    for (ParticleSystem system in _particles) {
+      if (active) {
+        actions.run(
+          new ActionTween((a) => system.opacity = a, system.opacity, 1.0, 2.0
+        ));
+      } else {
+        actions.run(
+          new ActionTween((a) => system.opacity = a, system.opacity, 0.0, 0.5
+        ));
+      }
+    }
+  }
+}
```

Comment:
there's some commonality between Snow and Rain that you can probably factor out into a Precipitation class